### PR TITLE
Set preferences in User model

### PR DESCRIPTION
**Pull request by abhiyaantrix**
- Set defaults
 - Validate preferences
 - Add serializer for JSONB column
 - Add specs

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -22,21 +22,32 @@
 #
 class User < ApplicationRecord
 
+  # TODO: Move this to Global Application Config class
+  THEMES ||= %w[system dark light].freeze
```
Comment (arnabry11) [inline app/models/user.rb]: Why are we using pipes here? Constants get loaded when we boot up the application

File: app/models/user.rb [hunk 2]
```diff
@@ -22,21 +22,32 @@
...
+
+  after_initialize :set_default_preferences, if: :new_record?
   before_save :sanitize_names
 
+  serialize :preferences, coder: HashSerializer
+  store_accessor :preferences, :theme, prefix: :preference
+  validates :preferences, preferences: true
```
Comment (arnabry11) [inline app/models/user.rb]: This line looks a bit weird but its fine. Can't think of a better name

File: app/models/user.rb [hunk 3]
```diff
@@ -22,21 +22,32 @@
...
+
   validates :email, presence: true, uniqueness: true
```
Comment (arnabry11) [inline app/models/user.rb]: We also need to validate email with email regex. Also use case_insensitive uniqueness.

**Review by arnabry11 — DISMISSED** [review]

File: app/models/user.rb (hunk 3 above)
Comment (arnabry11) [reply app/models/user.rb]: can be done in separate PR

File: app/models/user.rb (hunk 2 above)
Comment (abhiyaantrix) [reply app/models/user.rb]: Agree looks strange but that's how it is done with `EachValidator` unfortunately

https://guides.rubyonrails.org/active_record_validations.html#performing-custom-validations

File: app/models/user.rb (hunk 3 above)
Comment (abhiyaantrix) [reply app/models/user.rb]: Yes let me open a follow up PR for this, thanks

Comment (abhiyaantrix) [reply app/models/user.rb]: https://github.com/abhiyaantrix/event-radar/pull/19

**Review by arnabry11 — DISMISSED** [review]

**Pull request merged**